internal/api/databases.go

```diff
@@ -53,3 +56,40 @@ func handleGetDatabases(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 	outputJSON(c, w, databases)
 }
+
+// handleUpdateDatabase responds to PUT /api/database/{database}, updating the
+// database configuration values.
+func handleUpdateDatabase(c *Context, w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	databaseID := vars["database"]
+	c.Logger = c.Logger.WithField("database", databaseID)
+
+	database, status, unlockOnce := lockDatabase(c, databaseID)
+	if status != 0 {
+		w.WriteHeader(status)
+		return
+	}
+	defer unlockOnce()
+
+	patchDatabaseRequest, err := model.NewPatchDatabaseRequestFromReader(r.Body)
+	if err != nil {
+		c.Logger.WithError(err).Error("failed to decode request")
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	if patchDatabaseRequest.Apply(database) {
```

Comment:
I think it would be useful to indicate to the user whether the patch has been applied or not, or perhaps even return an error when trying to change the MaxInstallationPerDatabase value for DB other than PgBouncer one. It is all good while we remember it but might lead to some confusion in the future 🙂

Reply:
Yeah, this is a good idea and one I was also wondering if we should do someday. This type of patch logic is very common so I am thinking we design a general improvement and make sure all API endpoints get switched over at the same time.